internal/server/statvar/statvar_search.go

```diff
@@ -105,6 +124,43 @@ func filter(
 	return result
 }
 
+func groupStatVars(svList []*pb.EntityInfo, svgList []*pb.EntityInfo, parentMap map[string][]string) ([]*pb.EntityInfo, []*pb.SearchStatVarResponse_SearchResultSVG) {
+	// Create a map of svg id to svg search result
+	svgMap := map[string]*pb.SearchStatVarResponse_SearchResultSVG{}
+	for _, svg := range svgList {
+		svgMap[svg.Dcid] = &pb.SearchStatVarResponse_SearchResultSVG{
+			Dcid:     svg.Dcid,
+			Name:     svg.Name,
+			StatVars: []*pb.EntityInfo{},
+		}
+	}
+	resultSv := []*pb.EntityInfo{}
+	// Iterate through the list of stat vars. If a stat var has a parent svg that
+	// is a result, add the stat var to that svg result. If not, add the stat var
+	// to the result list of stat vars
+	for _, sv := range svList {
+		isGrouped := false
+		if parents, ok := parentMap[sv.Dcid]; ok {
+			for _, parent := range parents {
```

Comment:
I wonder should sort the parents here so stat var is always added to the top parent.

Reply:
I think that's a good idea, done